doc/source/rllib/rllib-learner.rst

```diff
@@ -0,0 +1,459 @@
+
+    learner.build() # needs to be called on the learner before calling any functions
+
+
+.. tab-set::
+        
+    .. tab-item:: Updating a LearnerGroup
+
+        .. testcode::
+            :skipif: True
+
+            # This is a blocking update
+            results = learner_group.update(DUMMY_BATCH)
+
+            # This is a non-blocking update. The results are returned in a future
+            # call to `async_update`
+            results = learner_group.async_update(DUMMY_BATCH)
+
+            # This is an additional non-gradient based update.
+            learner_group.additional_update(**ADDITIONAL_UPDATE_KWARGS)
+
+        When updating a `LearnerGroup` you can perform blocking or async updates on batches of data. Async updates are necessary for implementing async algorithms such as APPO/IMPALA.
+        You can perform non-gradient based updates using `additional_update`.
+
+    .. tab-item:: Updating a Learner
+
+        .. testcode::
+            :skipif: True
+
+            # This is a blocking update.
+            result = learner.update(DUMMY_BATCH)
+
+            # This is an additional non-gradient based update.
+            learner_group.additional_update(**ADDITIONAL_UPDATE_KWARGS)
+
+        When updating a `Learner` you can only perform blocking updates on batches of data.
+        You can perform non-gradient based updates using `additional_update`.
+    
+
+Getting and setting state
+-------------------------
+
+.. tab-set::
+    
+    .. tab-item:: Getting and Setting State for a LearnerGroup
+
+        .. testcode::
+            :skipif: True
+
+            # module weights and optimizer states
+            state = learner_group.get_state()
+            learner_group.set_state(state)
+
+            # just module weights
+            weights = learner_group.get_weights()
+            learner_group.set_weights(weights)
+
+        Set/get the state dict of all learners through learner_group via `set_state` or 
+        `get_state`. This includes all states including both neural network weights, 
+        and optimizer states on each learner.
+    
+    .. tab-item:: Getting and Setting State for a Learner
+
+        .. testcode::
+            :skipif: True
+
+            # module weights and optimizer states
+            state = learner.get_state()
+            learner.set_state(state)
+
+            # just module weights
+            weights = learner.module.get_state()
+            learner.module.set_state(weights)
+
+        You can set and get the weights of a `Learner` using `set_state` and `get_state` .
+        For setting or getting only RLModule weights (without optimizer states), use `set_weight` or `get_weight` API.
+
+
+.. testcode::
+	:hide:
+    :skipif: True
+
+	import shutil
+	import tempfile
+
+	LEARNER_CKPT_DIR = str(tempfile.TemporaryDirectory())
+	LEARNER_GROUP_CKPT_DIR = str(tempfile.TemporaryDirectory())
```

Comment:
What does this tell me? The snippet seems out of context to me

Reply:
This code is hidden to prevent users from seeing it in the guide.

Reply:
it was not hiding properly for some reason. I just removed it instead.